src/util/searchUtil.jsx

```diff
@@ -0,0 +1,22 @@
+//Function for filtering posts by title, community, text, or upi when user inputs text in the search bar, else all posts will be shown
+export function searchItem(
```

Comment:
This function should be refactored to be a custom hook. It should contain the state (searchQuery and filteredResults), and the updating logic should be in a useCallback.

It can return an object of any variables needed by the caller.